### PR TITLE
[BugFix] throw NPE when add a new partition to a lake table, then restart fe and drop this partition

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
CREATE TABLE `t` (
  `k` int(11) NOT NULL COMMENT "",
  `v` int(11) NOT NULL COMMENT ""
) ENGINE=starrocks
DUPLICATE KEY(`k`)
PARTITION BY RANGE(`v`) (
PARTITION p1 VALUES [("-2147483648"), ("0")),
PARTITION p2 VALUES [("0"), ("10")),
PARTITION p3 VALUES [("10"), ("20")))
DISTRIBUTED BY HASH(`k`) BUCKETS 5
PROPERTIES ("replication_num" = "1");

alter table t add partition p4 VALUES [("20"), ("30")) DISTRIBUTED BY HASH(`k`) buckets 5;

restart FE

alter table t drop partition p4;

Expected success，but it reported  "ERROR 1064 (HY000): Unexpected exception: null"

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|4|6|66.67%|[1564, 1565]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|

**Review by sduzh — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 1]
```diff
@@ -1423,6 +1423,7 @@ public void readFields(DataInput in) throws IOException {
 
     @Override
     public void gsonPostProcess() throws IOException {
+        super.gsonPostProcess();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:1426]: remove` implement GsonPostProcessable` in OlapTable

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|

**Pull request merged**